src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -57,9 +57,9 @@ await CreateBackupAsync(project,
                 projectJsonFilePath);
         }
 
-        private static async Task MigrateDependenciesAsync(BuildIntegratedNuGetProject project, PackageSpec packageSpec)
+        private static async Task MigrateDependenciesAsync(BuildIntegratedNuGetProject project, PackageSpecProjectJsonMigrationCandidate packageSpec)
         {
-            if (packageSpec.TargetFrameworks.Count > 1)
+            if (packageSpec.TargetFrameworkDependencies.Length > 1)
```

Comment:
These are not equivalent. 
The previous check was checking the amount of tfms, while this is changing the number of per tfm packages. 

If you follow my suggestion above, it should naturally fix this as well. 

The simplest way to say it is that I'd expect no changes in the implementation of the method, but just a change in the type like on line 60.